### PR TITLE
Add speech recognition tutorial

### PR DESCRIPTION
**Pull request by mthrok**
Add ASR example with torchaudio.

### CODE REVIEW

File: index.rst [hunk 1]
```diff
@@ -130,6 +130,13 @@ Welcome to PyTorch Tutorials
    :link: beginner/audio_preprocessing_tutorial.html
    :tags: Audio
 
+.. customcarditem::
+   :header: Automatic Speech Recognition with torchaudio
```
Comment (nateanl) [inline index.rst:134]: Rename ASR with Wav2vec2.0 ASR may attract more attentions from research community.

File: intermediate_source/speech_recognition_pipeline_tutorial.py [hunk 2]
```diff
@@ -0,0 +1,314 @@
+"""
+Speech Recognition with Torchaudio
+==================================
+
+**Author**: `Moto Hira <moto@fb.com>`__
+
+This tutorial shows how to perform speech recognition using using
```
Comment (nateanl) [inline intermediate_source/speech_recognition_pipeline_tutorial.py:7]: Could you add a colab logo and link to the colab tutorial?

Comment (mthrok) [reply intermediate_source/speech_recognition_pipeline_tutorial.py:7]: Colab link will be generated by the build process of the tutorial. See how [audio tutorial](https://pytorch.org/tutorials/beginner/audio_preprocessing_tutorial.html) has colab link but [the source](https://github.com/pytorch/tutorials/blob/master/beginner_source/audio_preprocessing_tutorial.py) does not.

Meanwhile here is the draft I used. https://colab.research.google.com/drive/1gPv7ugoJsigZrW_4wzhA6TZx7NUm2qQd#scrollTo=Dljo81T6pmKB

File: intermediate_source/speech_recognition_pipeline_tutorial.py [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+# 
+
+print(transcript)
+IPython.display.Audio(SPEECH_FILE)
+
+
+######################################################################
+# There are few remarks in decoding.
+# 
+# Firstly, the ASR model is fine-tuned using a loss function calle CTC.
```
Comment (carolineechen) [inline intermediate_source/speech_recognition_pipeline_tutorial.py:290]:
```suggestion
# Firstly, the ASR model is fine-tuned using a loss function called CTC.
```

File: intermediate_source/speech_recognition_pipeline_tutorial.py [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+# The detail of CTC loss is explained
+# `here <https://distill.pub/2017/ctc/>`__. In CTC a blank token (ϵ) is a
+# special token which represents a repetation of the previous symbol. In
+# decoding, these are simply ignored.
+# 
+# Secondly, as is explained in the feature extraction section, the
+# Wav2Vec2 model originated from ``fairseq`` has labels that are not used.
+# These also has to be ignored.
```
Comment (carolineechen) [inline intermediate_source/speech_recognition_pipeline_tutorial.py:298]:
```suggestion
# These also have to be ignored.
```

File: intermediate_source/speech_recognition_pipeline_tutorial.py [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+# Conclusion
+# ----------
+# 
+# In this tutorial, we looked at how to use ``torchaudio.pipeline`` to
+# perform acoustic feature extraction and speech recognition. Constructing
+# a model and getting the emission is as short as two lines.
+# 
+# ::
+# 
+#    model = buWAV2VEC2_ASR_BASE_960Hndle.get_model()
```
Comment (carolineechen) [inline intermediate_source/speech_recognition_pipeline_tutorial.py:312]: typo -- did you mean `bundle` or `torchaudio.pipelines.WAV2VEC2_ASR_BASE_960H`?

File: index.rst [hunk 6]
```diff
@@ -130,6 +130,13 @@ Welcome to PyTorch Tutorials
    :link: beginner/audio_preprocessing_tutorial.html
    :tags: Audio
 
+.. customcarditem::
+   :header: Automatic Speech Recognition with Wav2Vec2 in torchaudio
+   :card_description: Learn how to use torchaudio's pretrained models for building a speech recognition application.
+   :image: _static/img/thumbnails/cropped/torchaudio-asr.png
+   :link: intermediate_source/speech_recognition_pipeline_tutorial.html
+   :tags: Audio
+
```
Comment (carolineechen) [inline index.rst:139]: I think you also need to add `intermediate_source/speech_recognition_pipeline_tutorial.html` to the Audio section below

File: intermediate_source/speech_recognition_pipeline_tutorial.py [hunk 7]
```diff
@@ -0,0 +1,314 @@
...
```
Comment (mthrok) [inline intermediate_source/speech_recognition_pipeline_tutorial.py:312]:
```suggestion
#    model = torchaudio.pipelines.WAV2VEC2_ASR_BASE_960H.get_model()
```

File: intermediate_source/speech_recognition_pipeline_tutorial.py [hunk 8]
```diff
@@ -0,0 +1,314 @@
...
+######################################################################
+# Generating transcripts
+# ----------------------
+# 
+# From the sequence of label probabilities, now we want to generate
+# transcripts. The process to generate hypotheses is often called
+# “decoding”.
+# 
+# Decoding is more elaborate than simple classification because
+# classification at certain time step can be affected by surrounding
```
Comment (mthrok) [inline intermediate_source/speech_recognition_pipeline_tutorial.py:228]:
```suggestion
# decoding at certain time step can be affected by surrounding
```

File: intermediate_source/speech_recognition_pipeline_tutorial.py [hunk 9]
```diff
@@ -0,0 +1,314 @@
...
+
+
+######################################################################
+# Model can be constructed as following. This process will automatically
+# fetch the pre-trained weights and load it into the model.
+# 
+
+model = bundle.get_model().to(device)
+
+print(type(model))
```
[the remark this answers is not in the document]
Comment (mthrok) [reply intermediate_source/speech_recognition_pipeline_tutorial.py:108]: very good point.

Comment (mthrok) [inline intermediate_source/speech_recognition_pipeline_tutorial.py:108]:
```suggestion
print(model.__class__)
```

File: index.rst (hunk 6 above)
Comment (mthrok) [reply index.rst:139]: good catch. Thanks!

**Pull request merged**